### ISSUE TITLE
NullPointerException in QueryPartList.removeNulls with Java 11 and immutable collections

### ISSUE THREAD

**Issue by arlampin** [jOOQ/jOOQ#7991]
### Expected behavior and actual behavior:
Providing Java 10+ immutable collection (`List.of()` and the like) as select list should not throw an exception

### Steps to reproduce the problem (include example code if possible):
`context.select(List.of())`

It seems JDK's own immutable collections follow the spec strictly and throw NullPointerException even for `contains(null)` call.

```
        at java.util.Objects.requireNonNull(Objects.java:221) ~[?:?]
        at java.util.ImmutableCollections$AbstractImmutableList.indexOf(ImmutableCollections.java:169) ~[?:?]
        at java.util.ImmutableCollections$AbstractImmutableList.contains(ImmutableCollections.java:200) ~[?:?]
        at org.jooq.impl.QueryPartList.removeNulls(QueryPartList.java:211) ~[jooq-3.11.5.jar:?]
        at org.jooq.impl.QueryPartList.addAll(QueryPartList.java:199) ~[jooq-3.11.5.jar:?]
        at org.jooq.impl.SelectQueryImpl.addSelect(SelectQueryImpl.java:1743) ~[jooq-3.11.5.jar:?]
        at org.jooq.impl.SelectImpl.select(SelectImpl.java:260) ~[jooq-3.11.5.jar:?]
        at org.jooq.impl.DefaultDSLContext.select(DefaultDSLContext.java:2137) ~[jooq-3.11.5.jar:?]
```

See also #2145 

### Versions:

- jOOQ: 3.11.5
- Java: OpenJDK 11
- Database (include vendor):
- OS:
- JDBC Driver (include name if inofficial driver):

**Comment by lukaseder** [comment]
Thanks for your report. Indeed, it is regrettable that the collection API is so ambiguous about `null` values being allowed in collection subtypes. In any case, the possibility of this being prohibited is definitely enough to avoid calling `contains(null)`. The same functionality could be achieved by explicitly iterating the collection and checking for `null` values.

Or, maybe the feature isn't really needed anymore.

**Comment by lukaseder** [comment]
This is a regression that was introduced in jOOQ 3.0: #2145

It is now fixed in jOOQ 3.12. Backports scheduled to 3.11.6 (#8002) and 3.10.9 (#8003)

**Comment by arlampin** [comment]
Was this actually merged to 3.11.x branch? Release notes say yes, but the code doesn't seem to be there.

**Comment by lukaseder** [comment]
@arlampin : You're right, it's not there. Must have been an oversight during the merge process. Thanks for the pointer. I've reopened #8002
